webknossos/webknossos/skeleton/entities.py

```diff
@@ -0,0 +1,479 @@
+import colorsys
+import itertools
+from typing import Any, Dict, Generator, Iterator, List, Optional, Tuple, Union
+
+import attr
+import networkx as nx
+import numpy as np
+from icecream import ic
+
+import webknossos.skeleton.legacy as legacy_wknml
+from webknossos.skeleton.exporter import NMLExporter
+from webknossos.skeleton.legacy import NML as LegacyNML
+from webknossos.skeleton.legacy import Group as LegacyGroup
+from webknossos.skeleton.legacy import Tree as LegacyTree
+
+Vector3 = Tuple[float, float, float]
+Vector4 = Tuple[float, float, float, float]
+IntVector6 = Tuple[int, int, int, int, int, int]
+
+Number = Union[int, float]
+
+GroupOrGraph = Union["Group", "WkGraph"]
+
+nml_id_generator = itertools.count()
+
+
+def opt_vector3_as_float(
+    vec: Optional[Tuple[Number, Number, Number]]
+) -> Optional[Vector3]:
+    if vec is None:
+        return None
+    return (
+        float(vec[0]),
+        float(vec[1]),
+        float(vec[2]),
+    )
+
+
+def vector3_as_float(vec: Tuple[Number, Number, Number]) -> Vector3:
+    return (
+        float(vec[0]),
+        float(vec[1]),
+        float(vec[2]),
+    )
```

Comment:
I'm not sure what you mean? The function itself is not for the sake of typing, but for enforcing floats which is important for the roundtrip-test (nml -> skeleton -> nml) to succeed (otherwise, there are differences between floats and ints in the NML files). There's probably another way to fix this, but this seemed like an okay solution for now.